CIP-0015/CIP-0015.md

```diff
@@ -33,57 +33,88 @@ A voting key is simply an ED25519 key. How this key is created is up to the wall
 
 ### Associating stake with a voting key
 
+This method has been used since Fund 2.
+For future fund iterations, a new method making use of time-lock scripts may
+be introduced as described [below][future-development].
+
 Recall: Cardano uses the UTXO model so to completely associate a wallet's balance with a voting key (i.e. including enterprise addresses), we would need to associate every payment key to a voting key individually. Although there are attempts at this (see [CIP8](../CIP-0008/CIP-0008.md)), the resulting data structure is a little excessive for on-chain metadata (which we want to keep small)
 
 Given the above, we choose to only associate staking keys with voting keys. Since most Cardano wallets only use base addresses for Shelley wallet types, in most cases this should perfectly match the user's wallet.
 
 ### Registration metadata format
 
-A Catalyst registration transaction are a regular Cardano transaction with a specific transaction metadata associated with it
+A Catalyst registration transaction is a regular Cardano transaction with a specific transaction metadata associated with it.
 
-Notably, there should be three entries inside the metadata map:
+Notably, there should be four entries inside the metadata map:
 
-Voting key registration
+Voting registration example:
 ```
 61284: {
   // voting_key - CBOR byte array
   1: "0xa6a3c0447aeb9cc54cf6422ba32b294e5e1c3ef6d782f2acff4a70694c4d1663",
   // stake_pub - CBOR byte array
   2: "0xad4b948699193634a39dd56f779a2951a24779ad52aa7916f6912b8ec4702cee",
-  // address - CBOR byte array
-  3: "0x00588e8e1d18cba576a4d35758069fe94e53f638b6faf7c07b8abd2bc5c5cdee47b60edc7772855324c85033c638364214cbfc6627889f81c4"
+  // reward_address - CBOR byte array
+  3: "0x00588e8e1d18cba576a4d35758069fe94e53f638b6faf7c07b8abd2bc5c5cdee47b60edc7772855324c85033c638364214cbfc6627889f81c4",
+  // nonce
```

Comment:
the nonce cannot be reasonably verified by the signing user (it's just a number for them) and requires a stateful wallet/voting tools implementation which would need to track what nonce did the last voting key registration have.

I'd propose not discarding the slot number as the nonce value but rather changing it "min slot number" which would be required to be no higher than the slot number when the registration is submitted. That way anytime a user creates a voting registration transaction, they (or their wallet) can just set it to the current slot number and be confident there is no other active registration which would have a higher "nonce" value

Reply:
Since the only requirement for a nonce is that it should be monotonically rising, you can use slot numbers, or anything else you find convenient. There is no requirement to start from zero, increase nonce by a certain value every time, etc.

Reply:
If it becomes a monotonically increasing value (counter or slot number) then it should be called something other than nonce. The term nonce makes people assume it's random.

Reply:
IIRC Ethereum uses "nonce" for values that must increment as well, but we can rename this for clarity.

Reply:
In hw wallets' UI it's already called "Nonce" - not sure if we would have time to change it there before their respective releases, especially in case of Trezor, if we agree on some different name. Of course, it can be aligned in a later release

Reply:
We stand corrected :smiley: 

https://en.wikipedia.org/wiki/Cryptographic_nonce